packages/@aws-cdk/core/lib/private/refs.ts

```diff
@@ -156,11 +161,36 @@ function findAllReferences(root: IConstruct) {
 // export/import
 // ------------------------------------------------------------------------------------------------
 
+function createSsmParameter(consumer: Stack, reference: CfnReference): Intrinsic {
+  const exportingStack = Stack.of(reference.target);
+  const id = makeUniqueId([ JSON.stringify(exportingStack.resolve(reference)) ]);
+
+  // include the stack name, SSM parameters are not stack-local.
+  // use SSM hierarchy format - https://docs.aws.amazon.com/systems-manager/latest/userguide/sysman-paramstore-su-organize.html
+  const parameterName = `/stacks/${exportingStack.stackName}/${id}`;
+
+  new SsmStringParameter(exportingStack, `SSMExport${id}`, {
+    name: parameterName,
+    value: Token.asString(reference),
+    description: `[cdk] exported from stack "${exportingStack.stackName}" for use as parameter in a different stack`,
+  });
+
+  const parameterId = `SsmParameterValue:${parameterName}:C96584B6-F00A-464E-AD19-53AFF4B05118`;
+  let cfnparameter = consumer.node.tryFindChild(parameterId);
```

Comment:
Put the `as CfnElement | undefined` cast here. Putting it below in the `Ref` feels silly.